gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/SSOCookieFederationFilter.java

```diff
@@ -188,8 +194,25 @@ private void sendRedirectToLoginURL(HttpServletRequest request, HttpServletRespo
   }
 
   @Override
-  protected void handleValidationError(HttpServletRequest request, HttpServletResponse response,
-                                       int status, String error) throws IOException {
+  protected void handleValidationError(HttpServletRequest request, HttpServletResponse response, int status, String error) throws IOException {
+    if (error != null && error.startsWith("Token") && error.endsWith("disabled")) {
+      LOGGER.invalidSsoCookie();
+      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+      removeAuthenticationToken(request, response);
+      SessionInvalidators.KNOX_SSO_INVALIDATOR.getSessionInvalidators().forEach(sessionInvalidator -> {
+        sessionInvalidator.onAuthenticationError(request, response);
+      });
+      final boolean doGlobalLogout = request.getAttribute("doGlobalLogout") == null ? false
```

Comment:
`doGlobalLogout` should be a constant, it is used in `Pac4jDispatcherFilter` too.

Reply:
Ack.

Reply:
Fixed.